services/management/ia/channels_service.go

```diff
@@ -17,15 +17,242 @@
 package ia
 
 import (
+	"context"
+
 	iav1beta1 "github.com/percona/pmm/api/managementpb/ia"
+	"github.com/pkg/errors"
+
+	"github.com/percona/pmm-managed/models"
+	"github.com/percona/pmm-managed/services/ia"
 )
 
+// ChannelsService represents integrated alerting channels API.
 type ChannelsService struct {
-	iav1beta1.UnimplementedChannelsServer // TODO remove
+	ia alertingService
+}
+
+// NewChannelsService creates new channels API service.
+func NewChannelsService(ia *ia.Service) *ChannelsService {
+	return &ChannelsService{
+		ia: ia,
+	}
+}
+
+// ListChannels returns list of available channels.
+func (s *ChannelsService) ListChannels(ctx context.Context, request *iav1beta1.ListChannelsRequest) (*iav1beta1.ListChannelsResponse, error) {
+	channels, err := s.ia.ListChannels()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get notification channels")
+	}
+
+	res := make([]*iav1beta1.Channel, len(channels))
+	for i, channel := range channels {
+		c := &iav1beta1.Channel{
+			ChannelId: channel.ID,
+			Summary:   channel.Summary,
+			Disabled:  channel.Disabled,
+		}
+
+		switch channel.Type {
+		case models.Email:
+			config := channel.EmailConfig
+			c.Channel = &iav1beta1.Channel_EmailConfig{
+				EmailConfig: &iav1beta1.EmailConfig{
+					SendResolved: config.SendResolved,
+					To:           config.To,
+				},
+			}
+		case models.PagerDuty:
+			config := channel.PagerDutyConfig
+			c.Channel = &iav1beta1.Channel_PagerdutyConfig{
+				PagerdutyConfig: &iav1beta1.PagerDutyConfig{
+					SendResolved: config.SendResolved,
+					RoutingKey:   config.RoutingKey,
+					ServiceKey:   config.ServiceKey,
+				},
+			}
+		case models.Slack:
+			config := channel.SlackConfig
+			c.Channel = &iav1beta1.Channel_SlackConfig{
+				SlackConfig: &iav1beta1.SlackConfig{
+					SendResolved: config.SendResolved,
+					Channel:      config.Channel,
+				},
+			}
+		case models.WebHook:
+			config := channel.WebHookConfig
+			c.Channel = &iav1beta1.Channel_WebhookConfig{
+				WebhookConfig: &iav1beta1.WebhookConfig{
+					SendResolved: config.SendResolved,
+					Url:          config.URL,
+					HttpConfig:   convertModelToHTTPConfig(config.HTTPConfig),
+					MaxAlerts:    config.MaxAlerts,
+				},
+			}
+		default:
+			return nil, errors.Wrapf(err, "unknown notification channel type %s", channel.Type)
+		}
+
+		res[i] = c
+	}
+
+	return &iav1beta1.ListChannelsResponse{Channels: res}, nil
+}
+
+// AddChannel adds new notification channel.
+func (s *ChannelsService) AddChannel(ctx context.Context, req *iav1beta1.AddChannelRequest) (*iav1beta1.AddChannelResponse, error) {
+	params := &models.CreateChannelParams{
+		Summary:  req.Summary,
+		Disabled: req.GetDisabled(),
+	}
+
+	if emailConf := req.GetEmailConfig(); emailConf != nil {
```

Comment:
same as https://github.com/percona/pmm-managed/pull/532/files#r531430089

Reply:
Here we can hide different requests with some interface, but we preparing different parameters structures. So we can't unify that code.